### PR TITLE
Alphabetize JSX props in the frontend

### PR DESCRIPTION
**Pull request by paulmelnikow**
I'm game to give this a shot.

Ref #2701

### CODE REVIEW

File: frontend/components/common.spec.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ describe('Common modules', function() {
 
     it('contains a link to the image', function() {
       const wrapper = render(<common.Badge src="/badge/foo-bar-blue.svg" />)
-      expect(wrapper.html()).to.contain('<img src="/badge/foo-bar-blue.svg"')
```
Comment (calebcartwright) [inline frontend/components/common.spec.js]: Was the inclusion of the `img` tag unnecessary before, or if not, what's the thinking behind removing it (asking mostly for my own education on testing in React)

Comment (paulmelnikow) [reply frontend/components/common.spec.js]: Before I think it was `` and now I  think it's flipped around and turned into `` so the old version won't match anymore. This is now a really confusing assertion; why don't I figure out the full source of the `` tag and use that.

**Comment by paulmelnikow** [comment]
> Is this something we can get prettier to automatically update/enforce for us?

Appears not (prettier/prettier#323), though it is handled here 99% automatically by eslint. I think all but one of these was handled by `npm run lint -- --fix`.

File: .eslintrc-frontend.yml [hunk 2]
```diff
@@ -22,3 +22,4 @@ rules:
   no-console: 'error'
 
   import/extensions: ['error', 'never', { 'json': 'always', 'yml': 'always' }]
+  react/jsx-sort-props: 'error'
```
Comment (paulmelnikow) [inline .eslintrc-frontend.yml]: This is the addition of the linter rule.

File: frontend/components/common.spec.js (hunk 1 above)
Comment (calebcartwright) [reply frontend/components/common.spec.js]: 👍

**Pull request merged**